### ISSUE TITLE
HttpClientModule has broken compatibility with IE 9 ~ 11

### ISSUE THREAD

**Issue by gigioSouza** [angular/angular#19350]
## I'm submitting a...
[x] Bug report

## Current behavior
In older browsers as IE 9 ~ 11 when invoking .put() or .delete() methods from the HttpClient the browser throws SyntaxError.

## Expected behavior
It was expected to work just as the older HttpModule.

## Minimal reproduction of the problem with instructions
It can't be reproduced if tested on Edge emulators, only in the browsers IE 9, 10 and 11.

Browser:
[x] IE version 9, 10, 11

**Comment by gkalpak** [comment]
I tried to reproduce this but couldn't. Can you create a minimal reproduction (e.g. a repo we can clone and investigate)?

**Comment by gigioSouza** [comment]
@gkalpak I'm creating an app to reproduce it.

**Comment by gigioSouza** [comment]
@gkalpak I really tried to reproduce the error in a new project, I event copied my companie's project and renamed and deleted useless code for the reproduction but I failed. I couldn't reproduce the error.

But in the original project it persists. I hunted down this bug, I thought it'd be related to my interceptors configurations but it simply break until the response, so I really don't have a clue where is broken. I'd love to post the original project for the sake of resolving this issue, but I can't because of customer and sensitive data.
